src/Citeproc/Eval.hs

```diff
@@ -1162,18 +1162,30 @@ evalItem layout (position, item) = do
                       , IdentPMID  <$> (valToText =<< lookupVariable "PMID" ref)
                       , IdentURL   <$> (valToText =<< lookupVariable "URL" ref)
                       ]
+             let mburl = identifierToURL <$> mbident
+            
              -- hyperlink any titles in the output
-             let linkTitle target (Tagged TagTitle x) = Tagged (TagLink target) x
+             let linkTitle target (Tagged TagTitle x) = Linked LinkTitle target [x]
                  linkTitle _ x = x
              
-             -- when no links were rendered, hyperlink title, if it exists
-             usedLink <- gets stateUsedIdentifier
-
-             let xs' = case mbident of
-                         Nothing     -> xs
-                         Just ident  -> if usedLink
-                                           then xs
-                                           else fmap (transform (linkTitle ident)) xs
+             usedLink  <- gets stateUsedIdentifier
+             usedTitle <- gets stateUsedTitle
+             inBiblio  <- asks contextInBibliography 
+
+             -- when no links were rendered for a bibliography item, hyperlink
+             -- the title, if it exists, otherwise hyperlink the whole item
+             let xs' =
```

Comment:
I'll try to clean up this nested if/else later

Reply:
I removed one level of nesting, which I think is good enough unless you have other suggestions.  Other attempts to clean it up with `<|>` / `msum` etc I think made it harder to read and required defining some functions that wouldn't be used anywhere else.